Documentation Restructure (#1231)
NVIDIA/NeMo-Agent-Toolkit: docs/source/resources/contributing/index.md

## Developing with NeMo Agent toolkit

Refer to the [Get Started](../../get-started/installation.md) guide to quickly begin development.

## Documentation

All NeMo Agent toolkit documentation should be written in Markdown format.

The documentation located under the `docs/source` directory is included in the documentation builds, and is built using [Sphinx](https://www.sphinx-doc.org/en/master/) with the [MyST parser](https://myst-parser.readthedocs.io/en/latest/). Refer to `docs/README.md` for information on how to build the documentation.

In addition to this, each example should contain a `README.md` file that describes the example. This documentation is typically read using the GitHub browser interface or locally in a user's IDE. Example documentation should avoid any non-standard Markdown extensions to ensure compatibility.

### Documentation Categories

> The documentation in the `docs/source` directory is organized into the following categories:

Reviewer: ```suggestion
The documentation in the `docs/source` directory is organized into the following categories. When adding new documentation, ensure that it is added to the appropriate category and sub-category:
```

> * **About** (`about`) - This should be limited to an overview, key features, concepts and release notes documents.

Reviewer: ```suggestion
* **About** (`about`) - This should be limited to an overview, key features, concepts, and release notes documents.
```

> * **Get Started** (`get-started`) - introductory documentation to help users get started with NeMo Agent toolkit.

Reviewer: ```suggestion
* **Get Started** (`get-started`) - Introductory documentation to help users get started with NeMo Agent toolkit.
```

>   * **Tutorials** (`get-started/tutorials`) - step-by-step guides to help users learn how to use the toolkit.

Reviewer: ```suggestion
  * **Tutorials** (`get-started/tutorials`) - Step-by-step guides to help users learn how to use the toolkit to execute specific use cases.
```

> * **Build** (`build-workflows`) - documentation on how to build workflows and workflow configuration. This includes information on the types of components (for example LLMs, and agents, etc.) that can be used as part of a workflow. However documentation of specific component implementations (for example ReAct Agent, and the Code Execution tool) should be located in the Components category. Information about adding a remote MCP tool or A2A agent should be located here, however serving a workflow via MCP or A2A protocols should be located in the Run category.

Reviewer: ```suggestion
* **Build** (`build-workflows`) - Documentation on how to build workflows and workflow configuration. This includes information on the types of components (for example LLMs and agents) that can be used as part of a workflow. However, documentation of specific component implementations (for example ReAct Agent, and the Code Execution tool) should be located in the Components category. Information about adding a remote MCP tool or A2A agent should be located here, however serving a workflow using MCP or A2A protocols should be located under the "Run Workflows" label.
```

Reviewer: Last sentence is really confusing. Where do we want to add information about adding MCP too or A2A agent? Not sure where "here" is.

> * **Run** (`run-workflows`) - documentation on how to run workflows, including observing workflows and publishing workflows with MPC and A2A protocols.

Reviewer: ```suggestion
* **Run** (`run-workflows`) - Documentation on how to run workflows, including observing workflows and publishing workflows with MPC and A2A protocols.
```

> * **Improve** (`improve-workflows`) - documentation on how to improve workflows such as evaluation, and optimization guides.

Reviewer: ```suggestion
* **Improve** (`improve-workflows`) - Documentation on how to improve workflows such as evaluation and optimization guides.
```

> * **Components** (`components`) - documentation on specific component implementations such as agents, tools, and connectors.

Reviewer: ```suggestion
* **Components** (`components`) - Documentation on specific component implementations such as agents, tools, and connectors.
```

> * **Extend** (`extend`) - documentation on how to write custom components, along with how to test those components. The documentation here should help users extend the toolkit to meet their needs, and while there is some overlap, this documentation should not focus on contributing to the core NeMo Agent toolkit library.

Reviewer: ```suggestion
* **Extend** (`extend`) - Documentation on how to write custom components, along with how to test those components. The documentation here should help users extend the toolkit to meet their needs, and while there is some overlap, this documentation should not focus on contributing to the core NeMo Agent toolkit library.
```

> * **Reference** (`reference`) - This documentation should be limited to documenting the Python API, REST API and CLI.

Reviewer: ```suggestion
* **Reference** (`reference`) - This documentation should be limited to documenting the Python API, REST API, and CLI.
```

> * **Resources** (`resources`) - documentation for the project as a whole such as contributing guidelines, code of conduct, licensing, and FAQs.

Reviewer: ```suggestion
* **Resources** (`resources`) - Documentation for the project as a whole such as contributing guidelines, code of conduct, licensing, and FAQs.
```

>   * **Contributing** (`resources/contributing`) - documentation on how to contribute to the NeMo Agent toolkit project including setting up a development environment, and running tests.

Reviewer: ```suggestion
  * **Contributing** (`resources/contributing`) - Documentation on how to contribute to the NeMo Agent toolkit project including setting up a development environment and running tests.
```

> When adding new documentation, ensure that it is added to the appropriate category and sub-category.

Reviewer: Can move this to before the list of categories.

Author: This line actually already is prior to the list, it remains here as a type-o.
